### PR TITLE
Remove GDPR from the "users" namespace

### PR DESCRIPTION
**Pull request by Ridhwana**
## What type of PR is this? (check all applicable)

- [x] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

Remove the `users` namespace for GDPR Delete Requests. 

**Context:**
In an upcoming PR we will be placing the gdpr_delete_requests and invitations route under the member manage scope. 

Prior to this PR, the urls looked as follows: 
- Invitations: http://localhost:3000/admin/invitations
- GDPR: http://localhost:3000/admin/users/gdpr_delete_requests

Now you can see that there is a bit of inconsistency above with one of these routes being namespaced under Users and the other not. My initial reaction was to namespace the  invitations under Users. However on further reflection I've decided to remove the User namespace on GDPR instead for the following reasons: 

1. Previously these tabs lived on the Users page, hence a Users namespace made more sense. In an upcoming PR that will be moved to the sidebar, hence perhaps they don't need to be so closely correlated anymore. 
2. In the sidebar we will be grouping these routes under a scope called `:member_manager`, hence if we kept the users namespace the URL would have read `http://localhost:3000/admin/member_manager/users/gdpr_delete_requests`, with `member_manager` in the URL `/users` seems redundant. 

Screenshot 2022-05-02 at 09 30 20

## Related Tickets & Documents

- Related Issue #https://github.com/forem/forem/issues/17454
- Closes #https://github.com/forem/forem/issues/17454

## QA Instructions, Screenshots, Recordings

1. I've tested and checked for missed `gdpr_delete_requests` namespace references, but a second set of eyes wouldn't harm anyone :)
2. Play around on with the page, delete a user and then delete the GDPR and ensure it all still works with no errors. 

### UI accessibility concerns?

None, moved some code around.

## Added/updated tests?

- [x] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [ ] I've updated the [Developer Docs](https://developers.forem.com) or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
- [ ] This PR changes the Forem platform and our documentation needs to be
      updated. I have filled out the
      [Changes Requested](https://github.com/forem/admin-docs/issues/new?assignees=&labels=&template=changes_requested.md)
      issue template so Community Success can help update the Admin Docs
      appropriately.
- [ ] I've updated the README or added inline documentation
- [ ] I've added an entry to
      [`CHANGELOG.md`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
- [ ] I will share this change in a [Changelog](https://forem.dev/t/changelog)
      or in a [forem.dev](http://forem.dev) post
- [x] I will share this change internally with the appropriate teams
- [ ] I'm not sure how best to communicate this change and need help
- [ ] This change does not need to be communicated, and this is why not: _please
      replace this line with details on why this change doesn't need to be
      shared_

## [optional] What gif best describes this PR or how it makes you feel?

alt_text

### CODE REVIEW

File: app/views/admin/users/_menu.html.erb [hunk 1]
```diff
@@ -7,10 +7,10 @@
       <%= link_to "Invitations", admin_invitations_path, class: "crayons-navigation__item #{'crayons-navigation__item--current' if params[:controller] == 'admin/invitations'}", "aria-current": ("page" if params[:controller] == "admin/invitations") %>
     </li>
     <li>
-      <%= link_to admin_users_gdpr_delete_requests_path, class: "crayons-navigation__item #{'crayons-navigation__item--current' if params[:controller] == 'admin/users/gdpr_delete_requests'}", "aria-current": ("page" if params[:controller] == "admin/users/gdpr_delete_requests") do %>
+      <%= link_to admin_gdpr_delete_requests_path, class: "crayons-navigation__item #{'crayons-navigation__item--current' if params[:controller] == 'admin/users/gdpr_delete_requests'}", "aria-current": ("page" if params[:controller] == "admin/users/gdpr_delete_requests") do %>
```
Comment (aitchiss) [inline app/views/admin/users/_menu.html.erb:10]:
```suggestion
      <%= link_to admin_gdpr_delete_requests_path, class: "crayons-navigation__item #{'crayons-navigation__item--current' if params[:controller] == 'admin/gdpr_delete_requests'}", "aria-current": ("page" if params[:controller] == "admin/gdpr_delete_requests") do %>
```
I think this partial is set for removal so feel free to ignore if so 😄

File: app/views/admin/users/index/_tabs.html.erb [hunk 2]
```diff
@@ -7,10 +7,10 @@
       <%= link_to "Invited Members", admin_invitations_path, class: "crayons-navigation__item #{'crayons-navigation__item--current' if params[:controller] == 'admin/invitations'}", "aria-current": ("page" if params[:controller] == "admin/invitations") %>
     </li>
     <li>
-      <%= link_to admin_users_gdpr_delete_requests_path, class: "crayons-navigation__item #{'crayons-navigation__item--current' if params[:controller] == 'admin/users/gdpr_delete_requests'}", "aria-current": ("page" if params[:controller] == "admin/users/gdpr_delete_requests") do %>
+      <%= link_to admin_gdpr_delete_requests_path, class: "crayons-navigation__item #{'crayons-navigation__item--current' if params[:controller] == 'admin/users/gdpr_delete_requests'}", "aria-current": ("page" if params[:controller] == "admin/users/gdpr_delete_requests") do %>
```
Comment (aitchiss) [inline app/views/admin/users/index/_tabs.html.erb:10]:
```suggestion
      <%= link_to admin_gdpr_delete_requests_path, class: "crayons-navigation__item #{'crayons-navigation__item--current' if params[:controller] == 'admin/gdpr_delete_requests'}", "aria-current": ("page" if params[:controller] == "admin/gdpr_delete_requests") do %>
```
This tweak should restore the current tab highlighting

**Review by aitchiss — APPROVED** [review]
I've left a small change request, but other than that looks good to go 🚀

**Review by joshpuetz — APPROVED** [review]

**Review by juliannatetreault — APPROVED** [review]
🌟

File: app/views/admin/users/_menu.html.erb (hunk 1 above)
Comment (Ridhwana) [reply app/views/admin/users/_menu.html.erb]: Merged the changes where this was removed ✅

File: app/views/admin/users/index/_tabs.html.erb (hunk 2 above)
Comment (Ridhwana) [reply app/views/admin/users/index/_tabs.html.erb]: Done ✅

**Pull request merged**